example/spec/parallel/discoverGranules/DiscoverGranulesS3FailureSpec.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const pWaitFor = require('p-wait-for');
+const { randomString } = require('@cumulus/common/test-utils');
+const {
+  buildAndExecuteWorkflow,
+  loadCollection,
+  loadProvider
+} = require('@cumulus/integration-tests');
+const {
+  createCollection, deleteCollection
+} = require('@cumulus/integration-tests/api/collections');
+const { getExecution } = require('@cumulus/integration-tests/api/executions');
+const {
+  createProvider, deleteProvider
+} = require('@cumulus/integration-tests/api/providers');
+const { loadConfig } = require('../../helpers/testUtils');
+
+describe('The DiscoverGranules workflow with a non-existent bucket', () => {
+  let beforeAllCompleted = false;
+  let collection;
+  let provider;
+  let workflowExecution;
+  let stackName;
+  let bucket;
+
+  beforeAll(async () => {
+    ({ stackName, bucket } = await loadConfig());
+
+    const testId = randomString();
+
+    // Create the provider
+    provider = await loadProvider({
+      filename: './data/providers/s3/s3_provider.json',
+      postfix: testId,
+      s3Host: randomString()
+    });
+    await createProvider(stackName, provider);
+
+    // Create the collection
+    const loadedCollection = await loadCollection({
+      filename: './data/collections/s3_MOD09GQ_006/s3_MOD09GQ_006.json',
+      postfix: testId
+    });
+
+    collection = {
+      ...loadedCollection,
+      provider_path: `cumulus-test-data/${testId}`
+    };
+    await createCollection(stackName, collection);
+
+    // Execute the DiscoverGranules workflow
+    workflowExecution = await buildAndExecuteWorkflow(
+      stackName,
+      bucket,
+      'DiscoverGranules',
+      collection,
+      provider
+    );
+
+    beforeAllCompleted = true;
+  });
+
+  afterAll(() =>
+    Promise.all([
+      deleteCollection(stackName, collection.name, collection.version),
+      deleteProvider(stackName, provider.id)
+    ]));
+
+  it('fails', () => {
+    if (!beforeAllCompleted) fail('beforeAll() failed');
+    else expect(workflowExecution.status).toEqual('FAILED');
+  });
+
+  it('can be fetched from the API', async () => {
+    if (!beforeAllCompleted) fail('beforeAll() failed');
+    else {
+      try {
+        await pWaitFor(
```

Comment:
I realize that the function is somewhat different, but we already have a `waitForModelStatus` function in `example/spec/helpers/apiUtils.js` that we're using throughout our tests. Is there some reason to not use it here? I kind of like that pattern better than doing `expect().nothing()` below

Reply:
To improve on `expect().nothing()`, I refactored this test to use `await expectAsync(...).toBeResolved();` instead.

As for using `waitForModelStatus`, my preference is to use the API to get the status of the execution (like I'm doing here) rather than query the DB directly. My understanding is that, as much as possible, the integration tests should be testing Cumulus from the perspective of an outside user. By using the API to get the status, we're using Cumulus's public interface rather than our behind-the-scenes knowledge of how Cumulus works.

Reply:
Note: that refactor hasn't been pushed yet.